### DISCUSSION TITLE
Suggest using "rosys.on_startup()" instead of "app.on_startup()"

### DISCUSSION THREAD

**Discussion by AndyPermaRobotics** [zauberzeug/rosys#45] [Ideas / Feature Requests]
Hello together,

I figured out that there are two callbacks, that a rosys script can use for startup initialization code.
One is from nicegui `app.on_startup()` and one is `rosys.on_startup()`. 
Actually `app.on_startup()` an `rosys.on_startup()` are connected internally (see [rosys.py](https://github.com/zauberzeug/rosys/blob/89d44efd9e40be125f00e614e6da7fcda137c197/rosys/rosys.py#L262)).

But there is (at least) one possible pitfall:
If I want to set `rosys.config.simulation_speed` programmatically, the change is overwritten if I do it in the `app.on_startup()` callback, because the nicegui callback is invoked before `persistence.restore()` is called in the startup() routine of rosys [see here](https://github.com/zauberzeug/rosys/blob/89d44efd9e40be125f00e614e6da7fcda137c197/rosys/rosys.py#L161C26-L161C26).

I stumbled over it then I tried to set the simulation_speed to 1.0 in the production mode, but the script used the persisted simulation_speed of my previous tests. 

I don't see a need to use `app.on_startup()` instead of `rosys.on_startup()` therefore I suggest to encourage the usage of `rosys.on_startup()` in the rosys examples and the documentation.

**Comment by rodja** [comment]
Sure. The examples should use `rosys.on_startup()`. And maybe we can even monkey patch `app.on_startup` to warn in a RoSys app?

**Reply by falkoschindler** [reply]
I tried monkey-patching `app.on_startup`. But NiceGUI is using it internally, e.g. for `ui.timer`, so we get warnings even if there is no `app.on_startup` in user-code. We could inspect the call stack, but that's getting pretty messy I think. I tend to simply fix the examples and documentation without monkey-patching the code.

**Comment by rodja** [comment]
@falkoschindler just created the nice PR https://github.com/zauberzeug/rosys/pull/46 to improve documentation in this regard.
